src/routes/Onboarding/utils.tsx

```diff
@@ -70,69 +86,94 @@ export function useDemoCode() {
   }, [editorManager.editorView, immediateState.type, overallState])
 }
 
-export function useNextClick(newStatus: string) {
+export function useNextClick(newStatus: OnboardingStatus) {
   const filePath = useAbsoluteFilePath()
   const navigate = useNavigate()
 
   return useCallback(() => {
+    if (!isOnboardingSubPath(newStatus)) {
+      return new Error(
+        `Failed to navigate to invalid onboarding status ${newStatus}`
+      )
+    }
```

Comment:
This is what will cause the E2E test to fail if a bad onboarding path is ever constructed. This is the only place in the code base that the onboarding status setting is set to something dynamic. If somehow the user sees this error state they will still be able to use the app by dismissing the onboarding.